### PR TITLE
Add Set-GcLogMetric cmdlet

### PR DESCRIPTION
**Pull request by quoctruong**
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-powershell/443)

### CODE REVIEW

File: Google.PowerShell.IntegrationTests/Logging/Logging.GcLogMetric.Tests.ps1 [hunk 1]
```diff
@@ -3,6 +3,13 @@ Install-GCloudCmdlets
 
 $project, $zone, $oldActiveConfig, $configName = Set-GCloudConfig
 
+function Get-DateTimeFromFilter($filter) {
```
Comment (iantalarico) [inline Google.PowerShell.IntegrationTests/Logging/Logging.GcLogMetric.Tests.ps1]: Can you add a method comment?

File: Google.PowerShell.IntegrationTests/Logging/Logging.GcLogMetric.Tests.ps1 [hunk 2]
```diff
@@ -3,6 +3,13 @@ Install-GCloudCmdlets
...
+    if ($filter -match ".*timestamp\s*[<>]=\s*`"(.*)`"") {
```
Comment (iantalarico) [inline Google.PowerShell.IntegrationTests/Logging/Logging.GcLogMetric.Tests.ps1]: Can you add a comment about what the regex matches?

File: Google.PowerShell.IntegrationTests/Logging/Logging.GcLogMetric.Tests.ps1 [hunk 3]
```diff
@@ -191,7 +196,7 @@ Describe "New-GcLogMetric" {
                 $metric.Filter | Should BeExactly "logName = `"projects/$project/logs/$logName`""
                 $metric.Description | Should BeExactly $description
             }
-        }
+        }
```
Comment (iantalarico) [inline Google.PowerShell.IntegrationTests/Logging/Logging.GcLogMetric.Tests.ps1]: nit: extra added white space

File: Google.PowerShell/Logging/GcLogMetricCmdlets.cs [hunk 4]
```diff
@@ -156,12 +122,17 @@ public class NewGcLogMetricCmdlet : GcLogEntryCmdletWithLogFilter
...
+        /// The description of the metric.
+        /// </para>
         /// </summary>
         [Parameter(Mandatory = false, Position = 1)]
         public string Description { get; set; }
 
+        /// <summary>
+        /// Given a log metric body, returns either a create or update request.
+        /// </summary>
+        protected abstract LoggingBaseServiceRequest<LogMetric> GetRequest(LogMetric logMetric);
```
Comment (iantalarico) [inline Google.PowerShell/Logging/GcLogMetricCmdlets.cs]: I think this comment is a little confusing.  This just returns a LoggingBaseServiceRequest right?  It's not two separate request types?

File: Google.PowerShell/Logging/GcLogMetricCmdlets.cs [hunk 5]
```diff
@@ -204,6 +167,133 @@ protected override void ProcessRecord()
...
+    /// The default project will be used to search for the metric if -Project is not used.
+    /// </para>
+    /// <example>
+    ///   <code>PS C:\> Set-GcLogMetric -MetricName "my-metric" -LogName "my-log"</code>
+    ///   <para>This command updates the metric "my-metric" to count the number of log entries in log "my-log".</para>
+    /// </example>
+    /// <example>
+    ///   <code>
+    ///   PS C:\> Set-GcLogMetric -MetricName "my-metric" `
+    ///                           -ResourceType gce_instance
```
Comment (iantalarico) [inline Google.PowerShell/Logging/GcLogMetricCmdlets.cs]: does gce_instance need to be in quotes?

**Comment by quoctruong** [comment]
@iantalarico PTAL

**Comment by quoctruong** [comment]
Thanks!

**Pull request merged**